common/server/stats.h

```diff
@@ -10,8 +10,11 @@
 
 #include "common/stats/provider.h"
 
-char *engine_default_prepare_stats_with_tag_mask(stats_t &&stats, int *len, const char *statsd_prefix, unsigned int tag_mask);
-char *engine_default_prepare_stats(stats_t &&stats, int *len, const char *statsd_prefix);
+constexpr int STATS_BUFFER_LEN = 1 << 20;
+
+char* get_engine_default_prepare_stats_buffer();
+std::pair<char *, int> engine_default_prepare_stats_with_tag_mask(stats_t &&stats, const char *stats_prefix, unsigned int tag_mask);
+char *engine_default_prepare_stats(stats_t &&stats, const char *statsd_prefix);
```

Comment:
`char *` here vs `char*` above — not in a style. We prefer `char *`. It's not the only place, btw.